### PR TITLE
feat: Autoload Twilio::REST and Twilio::HTTP

### PR DESCRIPTION
**Pull request by gmcgibbon**
Currently `Twilio::REST` and `Twilio::HTTP` are loaded with file globs. This works great, but it is slow. Especially for `Twilio::REST` which has lots of files. For applications using this gem, bundler will `require "twilio-ruby"` and load the entire gem at once. 

Ruby's `autoload` feature can help with this. Autoload defers loading modules until they are actually needed/referenced. By autoloading `Twilio::REST` and `Twilio::HTTP`, the require-time of this library is reduced from `~400ms` to `~25ms`. You can verify this by wrapping the require in your spec helper with a benchmark block:

```ruby
require "benchmark"
ms = 1000 * Benchmark.realtime do
  require 'twilio-ruby'
end
puts "Require took #{ms}ms"
```

This is what the gem require currently looks like in our application, with the majority of time taken by requiring `Twilio::REST`:
Screen Shot 2021-06-10 at 12 23 37 PM
After this patch, it did not show up in our profiles.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-ruby/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: lib/twilio-ruby.rb [hunk 1]
```diff
@@ -16,27 +16,20 @@
 require 'twilio-ruby/security/request_validator'
 require 'twilio-ruby/util/configuration'
 
-Dir[File.dirname(__FILE__) + '/twilio-ruby/http/**/*.rb'].sort.each do |file|
+Dir[File.join(__dir__, 'twilio-ruby/compatibility/**/*.rb')].sort.each do |file|
```
Comment (gmcgibbon) [inline lib/twilio-ruby.rb:19]: Also there's no files that match this glob. Maybe it can be removed.

Comment (thinkingserious) [reply lib/twilio-ruby.rb:19]: Agreed.

**Review by thinkingserious — COMMENTED** [review]
Thank you!! Please remove the uncessary glob.

**Review by eshanholtz — APPROVED** [review]
LGTM! Thank you for the contribution!

**Pull request merged**